internal/resources/util_fakes_test.go

```diff
@@ -34,6 +34,8 @@ type fakeClient struct {
 	clientGet func(context.Context, types.NamespacedName, rtclient.Object) error
 }
 
+type fakeSubResource struct{}
```

Comment:
Could you move this to directly above the methods? I think Go is OK with that and I was initially confused as to how fakeSubResource could meet the SubResourceClient interface until I scrolled much further down.